pkg/gpu/context.go

```diff
@@ -154,10 +155,13 @@ func (ctx *systemContext) getProcessMemoryMaps(pid int) (*kernel.ProcMapEntries,
 	return &maps, nil
 }
 
-func (ctx *systemContext) cleanupDataForProcess(pid int) {
+// removeProcess removes any data associated with a process from the system context.
+func (ctx *systemContext) removeProcess(pid int) {
```

Comment:
_nit:_ consider removing this wrapper and simply call directly `delete(ctx.pidMaps, pid)` instead